src/Microsoft.Framework.Configuration.EnvironmentVariables/EnvironmentVariables/EnvironmentVariablesConfigurationSource.cs

```diff
@@ -76,22 +76,26 @@ private static IEnumerable<DictionaryEntry> AzureEnvToAppEnv(DictionaryEntry ent
             {
                 prefix = CustomPrefix;
             }
-            else
+
+            key = key.Substring(prefix.Length).Replace("__", ":");
```

Comment:
The IsNullOrEmpty? Because if the prefix is empty we don't want to format the key, and Substring(0) does nothing.